tools/integration_tests/util/client/storage_client.go

```diff
@@ -222,3 +222,57 @@ func StatObject(ctx context.Context, client *storage.Client, object string) (*st
 	}
 	return attrs, nil
 }
+
+// UploadGcsObject uploads a local file to a specified GCS bucket and object.
+// Handles gzip compression if requested.
+func UploadGcsObject(ctx context.Context, localPath, bucketName, objectName string, uploadGzipEncoded bool) error {
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create storage client: %v", err)
+	}
+	defer client.Close()
+
+	// Create a writer to upload the object.
+	obj := client.Bucket(bucketName).Object(objectName)
+	w := obj.NewWriter(ctx)
+	defer func() {
+		err := w.Close()
+		if err != nil {
+			fmt.Printf("Failed to close GCS object gs://%s/%s: %v", bucketName, objectName, err)
```

Comment:
nit: ideally we should return this error and not just print it.

Reply:
We can't return anything in defer function and also returning a non nil error will result in test failure but not closing writer should not result in test failure.

Reply:
> We can't return anything in defer function

The way to return a value in defer function is to set a named returned value in it.

> also returning a non nil error will result in test failure but not closing writer should not result in test failure

Why is that ?

Reply:
Resolving this as per discussion.